managed/common.go

```diff
@@ -25,8 +25,28 @@ func getProjectId(accountId string, apiClient *openapiclient.APIClient) (project
 	return projectId, true, ""
 }
 
-func getAccountId(apiClient *openapiclient.APIClient) (accountId string, accountIdOK bool, errorMessage string) {
-	accountResp, resp, err := apiClient.AccountApi.ListAccounts(context.Background()).Execute()
+func getMemoryFromInstanceType(ctx context.Context, apiClient *openapiclient.APIClient, accountId string, cloud string, tier string, region string, numCores int32) (memory int32, memoryOK bool, errorMessage string) {
+	instanceResp, resp, err := apiClient.ClusterApi.GetSupportedInstanceTypes(context.Background()).AccountId(accountId).Cloud(cloud).Tier(tier).Region(region).Execute()
+	if err != nil {
+		b, _ := httputil.DumpResponse(resp, true)
+		return 0, false, string(b)
+	}
+	instanceData := instanceResp.GetData()
+	nodeConfigList, ok := instanceData[region]
+	if !ok || len(nodeConfigList) == 0 {
+		return 0, false, "No nodes configured for the given region."
+	}
+	for _, nodeConfig := range nodeConfigList {
+		if nodeConfig.GetNumCores() == numCores {
+			memory = nodeConfig.GetMemoryMb()
```

Comment:
I would add logs around this area (in particular before the `return` statements).

Reply:
Added the logs.